source/common/http/http2/codec_impl.cc

```diff
@@ -62,7 +62,7 @@ ConnectionImpl::StreamImpl::StreamImpl(ConnectionImpl& parent, uint32_t buffer_l
   }
 }
 
-ConnectionImpl::StreamImpl::~StreamImpl() { ASSERT(unconsumed_bytes_ == 0); }
```

Comment:
Oh yeah, I meant to call that out.

So it's possible for the entire connection to get torn down with open streams.  In this case there's really no way to consume. The only way to add this assert back in would be to loop through streams there before they're deleted in ~ConnectionImpl and zero it out and that seemed overkill.